core/pubnub_ntf_sync.c

```diff
@@ -130,6 +140,9 @@ enum pubnub_res pubnub_await(pubnub_t* pb)
         }
     }
     result = pb->core.last_result;
+    if (result != PNR_OK){
+        pbnc_tr_cxt_state_reset(pb);
+    }
```

Comment:
In pubnub_set_state() call, without knowing status PNR_OK, I am setting state to pb->core.state.
In the line you pointed, when the last result is NOT successful, I am resetting the set state so that there is no orphan state in the pubnub context.